### PR TITLE
enable NVPTX offload support in GCCcore 10.2.0 easyconfig

### PR DESCRIPTION
**Pull request by bartoldeman**
Requires ~~https://github.com/easybuilders/easybuild-easyblocks/pull/2235~~

### CODE REVIEW

**Comment by bartoldeman** [comment]
@boegelbot please test @ generoso EB_ARGS="--include-easyblocks-from-pr 2235"

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in this PR)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/a5afdb17eb72e05d5a400504808353b1 for a full test report.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/67c9b6fde5e7b5ff47425664c9483dcb for a full test report.

**Comment by bartoldeman** [comment]
Test report by @bartoldeman
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
build-node.computecanada.ca - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.7.7
See https://gist.github.com/07358daf5dd30270d751b158ea7597b7 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 2 out of 3 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/e367ac3a77e05aa8aee303b5920f787e for a full test report.

**Comment by bartoldeman** [comment]
@Micket seems you ran out of disk space
```
ld.gold: fatal error: lto1: No space left on device
ld.gold: fatal error: lto-dump: No space left on device
collect2: error: ld returned 1 exit status
```

**Comment by Micket** [comment]
There was a bunch of old failed builds on unrelated things taking up space on the ramdisk i use for building. Rebuilding.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.8.2003, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/f468248cc339269710f33c1504974f01 for a full test report.

File: easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb [hunk 1]
```diff
@@ -36,6 +40,8 @@ checksums = [
     'feced2d430dd5a97805fa289fed3fc8ff2b094c02d05287fd6133e7f1f0ec926',  # mpfr-4.1.0.tar.bz2
     '6985c538143c1208dcb1ac42cedad6ff52e267b47e5f970183a3e75125b43c2e',  # mpc-1.1.0.tar.gz
     '1a668ef92eb181a7c021e8531a3ca89fd71aa1b3744db56f68365ab0a224c5cd',  # isl-0.22.1.tar.bz2
+    '58dd9e3eaedf519360d92d84205c3deef0b3fc286685d1c562e245914ef72c66',  # newlib-3.3.0.tar.gz
+    'a25b6f7761bb61c0d8e2a183bcf51fbaaeeac26868dcfc015e3b16a33fe11705',  # 5f6f343.tar.gz
```
Comment (boegel) [inline easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb:44]:
```suggestion
    'a25b6f7761bb61c0d8e2a183bcf51fbaaeeac26868dcfc015e3b16a33fe11705',  # nvptx-tools-20180301.tar.gz

File: easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb [hunk 2]
```diff
@@ -17,13 +17,17 @@ source_urls = [
     'ftp://gcc.gnu.org/pub/gcc/infrastructure/',  # GCC dependencies
     'http://gcc.cybermirror.org/infrastructure/',  # HTTP mirror for GCC dependencies
     'http://isl.gforge.inria.fr/',  # original HTTP source for ISL
+    'ftp://sourceware.org/pub/newlib/', # for newlib
```
Comment (boegel) [inline easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb:20]:
```suggestion
    'https://sourceware.org/pub/newlib/',  # for newlib

File: easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb [hunk 3]
```diff
@@ -17,13 +17,17 @@ source_urls = [
...
+    'https://github.com/MentorEmbedded/nvptx-tools/archive', # for nvptx-tools
 ]
 sources = [
     'gcc-%(version)s.tar.gz',
     'gmp-6.2.0.tar.bz2',
     'mpfr-4.1.0.tar.bz2',
     'mpc-1.1.0.tar.gz',
     'isl-0.22.1.tar.bz2',
+    'newlib-3.3.0.tar.gz',
+    '5f6f343.tar.gz', # nvptx-tools
```
Comment (boegel) [inline easybuild/easyconfigs/g/GCCcore/GCCcore-10.2.0.eb:30]: @bartoldeman This should be renamed on download:

```suggestion
    {'download_filename': '5f6f343.tar.gz', 'filename': 'nvptx-tools-20180301.tar.gz'},

```

**Review by boegel — DISMISSED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3409.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/e3105cdcc13e3decf878f521bb2224e6 for a full test report.

**Comment by bartoldeman** [comment]
@boegel I applied all your changes or did I miss one by accident? It still says "1 change requested"

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node126.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/841463b12cc66fdb2d25836effa2df3d for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node375.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/76d6de6bd625c5284073111eed92df49 for a full test report.

**Review by lexming — APPROVED** [review]
LGTM

**Comment by lexming** [comment]
Going in, thanks @bartoldeman !

**Pull request merged**